libs/domain/search/pagination/pagination.component.ts

```diff
@@ -0,0 +1,59 @@
+import { resolve } from '@spryker-oryx/di';
+import {
+  ContentController,
+  ContentMixin,
+  defaultOptions,
+} from '@spryker-oryx/experience';
+import {
+  ProductListPageService,
+  ProductListQualifier,
+} from '@spryker-oryx/product';
+import { hydratable, signal } from '@spryker-oryx/utilities';
+import { html, LitElement } from 'lit';
+import { TemplateResult } from 'lit/development';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import urlJoin from 'url-join';
+import { PaginationOptions } from './pagination.model';
+
+@defaultOptions({ max: 3, enableControls: true })
+@hydratable('window:load')
+export class PaginationComponent extends ContentMixin<
+  PaginationOptions & ProductListQualifier
+>(LitElement) {
+  protected options$ = new ContentController(this).getOptions();
+
+  protected productListPageService = resolve(ProductListPageService);
+
+  protected $pagination = signal(
+    resolve(ProductListPageService).getPagination()
```

Comment:
we like to always bring resolved services in as named variables (`protected productListPageService`...)